pkgs/build-support/setup-hooks/patch-shebangs.sh

```diff
@@ -95,7 +94,7 @@ patchShebangs() {
                 rm "$timestamp"
             fi
         fi
-    done < <(find "$dir" -type f -perm -0100 -print0)
+    done < <(find "$@" -type f -perm -0100 -print0)
```

Comment:
This change, combined with the removal of line 43, means the function will fail clumsily if no arguments are passed to it. But considering that the old version silently and surprisingly ignores nonexistent files, this could be considered an improvement. ;) However, since it's a change in behavior, might it cause build failures? Should we let them happen? (Serious question.)

Reply:
What about we check if there are no arguments supplied in line 43?
```bash
  if [ $# -eq 0 ]; then
      echo "No arguments supplied to patchShebangs" >&2
      return 3141592  # Not sure which return code is best, maybe just 1
  fi
```

Reply:
I made it exit with 0 like it was before so that existing scripts don't break.